cmd/piecestore/main.go

```diff
@@ -61,7 +62,15 @@ func main() {
 					return argError.New(fmt.Sprintf("Path (%s) is a directory, not a file", c.Args().Get(1)))
 				}
 
-				_, err = pstore.Store(c.Args().Get(0), file, int64(fileInfo.Size()), 0, c.Args().Get(2))
+				dataFileChunk, err := pstore.StoreWriter(c.Args().Get(0), int64(fileInfo.Size()), 0, c.Args().Get(2))
```

Comment:
should you validate you got c.Args().Get(0) ?

Reply:
If the hash is invalid the data won't get stored by the server. Also this is just an example